gov/politeia/piclient/piclient.go

```diff
@@ -134,29 +103,24 @@ func RetrieveProposalByToken(client *http.Client, APIRootPath, token string) (*p
 		return nil, fmt.Errorf("retrieving %s proposal details failed: %v", token, err)
 	}
 
-	// Check if null proposal data was returned as part of the proposal details.
-	if bytes.Equal(data, nullProposalData) {
-		return nil, fmt.Errorf("invalid proposal with null data found for %s", token)
-	}
-
 	var proposal pitypes.Proposal
 	err = json.Unmarshal(data, &proposal)
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if null proposal data was returned as part of the proposal details.
+	if proposal.Data == nil {
+		return nil, fmt.Errorf("invalid proposal with null data found for %s", token)
```

Comment:
This feels like the only place a null pointer would wreck havoc if were to be returned by the API. 
Please validate this before I can proceed to made changes to the rest of PRs.

Reply:
Yeah, looks like it to me.  Thanks.